### PR TITLE
Suggestions based on Intellij's code analysis.

### PR DESCRIPTION
**Pull request by KoenDG**
This is a small PR I did in my spare time. I had Intellij's code analysis look at the code, then went over the results.

The biggest change is the use of a static inner class for AsyncCompletionHandler. As Intellij's description states:

>A static inner class does not keep an implicit reference to its enclosing instance. This prevents a common cause of memory leaks and uses less memory per instance of the class. 

Other than that, some small changes of which Intellij states they're more performant.

### CODE REVIEW

**Comment by marcospereira** [comment]
Hey @KoenDG,

Thank you for this. Could you please rebase against master to resolver the conflicts?

Best.

**Comment by marcospereira** [comment]
> EDIT2: Looks like they started running now. The 2.12.4/jdk9 test seems broken, but also seems broken on master, with the exact same output.

We had some problems with Travis cache. The build is green now after clearing the cache and running again.

**Comment by KoenDG** [comment]
@marcospereira ah yes, all green too here on the most recent build: https://travis-ci.org/playframework/play-ws/builds/392912248?utm_source=github_status&utm_medium=notification

File: play-ahc-ws-standalone/src/main/scala/play/api/libs/ws/ahc/cache/CacheableResponse.scala [hunk 1]
```diff
@@ -304,6 +304,6 @@ class CacheableHttpResponseBodyPart(chunk: Array[Byte], last: Boolean) extends H
   override def length(): Int = if (chunk != null) chunk.length else 0
 
   override def toString: String = {
-    s"CacheableHttpResponseBodyPart(last = $last, chunk size = ${chunk.size})"
+    s"CacheableHttpResponseBodyPart(last = $last, chunk size = $chunk.length)"
```
[the remark this answers is not in the document]
Comment (KoenDG) [reply play-ahc-ws-standalone/src/main/scala/play/api/libs/ws/ahc/cache/CacheableResponse.scala]: That was an automatic refactor by Intellij. Looks like bad behavior. I'll make a post about it on their bugtracker.

[the remark this answers is not in the document]
Comment (KoenDG) [reply play-ahc-ws-standalone/src/main/scala/play/api/libs/ws/ahc/cache/CacheableResponse.scala]: Fixed.

**Comment by KoenDG** [comment]
Okay, so I fixed that `${chunk.length}` bit, now the 2.12.4 with jdk build is failing at the others aren't.

```
...
[info] JsonRequestSpec
[error]   x stream with adhoc request filter
[error]    Timeout after 25 seconds (retries = 0, timeout = 5 seconds), timeFactor = 5 (AhcWSRequestFilterSpec.scala:69)
[info] AhcWSRequestFilterSpec
...
```

I've had this happen before on my first push. Is this test flaky or is there something actually wrong?

**Comment by KoenDG** [comment]
Forgot about this for a bit, merged in master and pushed. Build is passing now.

**Comment by marcospereira** [comment]
@KoenDG, yeah, and looks like none of those support Scala either.

**Pull request merged**